sonarts-core/src/utils/parser.ts

```diff
@@ -69,15 +70,23 @@ export function parseFile(filename: string): { sourceFile: ts.SourceFile; progra
   return { sourceFile, program };
 }
 
-export function createProgram(configFile: string): ts.Program {
-  const { options, files } = parseTsConfig(configFile);
+export function createProgram(configFile: string, projectRoot: string): ts.Program {
+  const { options, files } = parseTsConfig(configFile, projectRoot);
   const host = ts.createCompilerHost(options, true);
   return ts.createProgram(files, options, host);
 }
 
-export function parseTsConfig(tsConfig: string): { options: ts.CompilerOptions; files: string[] } {
-  const projectDirectory = path.dirname(tsConfig);
-  const config = ts.readConfigFile(tsConfig, ts.sys.readFile);
+export function parseTsConfig(tsConfig: string, projectRoot: string): { options: ts.CompilerOptions; files: string[] } {
+  let projectDirectory = projectRoot;
+  let config: {
+    config?: any;
```

Comment:
Will keep `any` despite being a "code smell" in SQ since it's a type coming from TS compiler API